### DISCUSSION TITLE
Change method display name

### DISCUSSION THREAD

**Discussion by Arthri** [thomhurst/TUnit#1280] [Q&A] [answered]
Is it possible to configure TUnit to transform `should_be_completeX12` to `should be complete!`, similar to how xUnit supports `methodDisplayOptions`?

**Comment by thomhurst** [comment]
`[DisplayName("should be completed!")]` on your test method

**Reply by Arthri** [reply]
I see, thanks for the response. Does TUnit support transforming it automatically? I feel that writing it separately can be redundant
```cs
[DisplayName("should be complete!")]
public async Task should_be_complete() { ... }
```

**Reply by thomhurst** [reply]
I can look at adding an attribute for you that would allow you to plug in your own custom logic. You could apply to the test, or class, or assembly

**Reply by Arthri** [reply]
Thanks for the offer, that would be nice yeah. I'm in no rush, however, so feel free to take as much time as needed

**Reply by thomhurst** [reply, marked as answer]
This is achievable in 0.4.1 by inheriting from the `DisplayNameFormatterAttribute` and plugging in custom logic.

Here's an example using the library Humanizer:

```csharp
[HumanizerDisplayName]
public class HumanizerDisplayNameTests
{
    [Test]
    public void This_test_name_is_formatted_nicely()
    {
        Console.WriteLine(TestContext.Current!.GetTestDisplayName());
    }
    
    public class HumanizerDisplayNameAttribute : DisplayNameFormatterAttribute
    {
        protected override string FormatDisplayName(TestContext testContext)
        {
            return testContext.TestDetails.TestName.Humanize();
        }
    }
}
```

**Reply by Arthri** [reply]
Awesome work. I appreciate your help. Although, I've noticed `DisplayNameAttribute` doesn't seem to work on methods with arguments(specifically ones populated with `MethodDataSource`)

Thanks for your work either way!

**Reply by thomhurst** [reply]
What is it doing?

**Reply by Arthri** [reply]
The display names of the test cases with arguments remain unchanged. They have the same name as the method
